src/OrleansServiceBus/Providers/Streams/EventHub/EventHubDataAdapter.cs

```diff
@@ -16,13 +17,49 @@ public struct CachedEventHubMessage
     {
         public Guid StreamGuid;
         public long SequenceNumber;
+        public DateTime EnqueueTimeUtc;
+        public DateTime DequeueTimeUtc;
         public ArraySegment<byte> Segment;
     }
 
     /// <summary>
+    /// Replication of EventHub EventData class, reconstructed from cached data CachedEventHubMessage
+    /// </summary>
+    [Serializable]
+    public class EventHubMessage
+    {
+        public EventHubMessage(ref CachedEventHubMessage cachedMessage)
```

Comment:
If this avoids an allocation, I'm all for it. I was just wondering if that's the case and if boxing is involved here.